src/test/suite/fixtures/recorded/languages/ruby/changeLambda4.yml

```diff
@@ -0,0 +1,23 @@
+languageId: ruby
+command:
+  version: 1
+  spokenForm: change lambda
+  action: clearAndSetSelection
+  targets:
+    - type: primitive
+      modifier: {type: containingScope, scopeType: anonymousFunction, includeSiblings: false}
+initialState:
+  documentContents: lambda { puts "Hi!" }
+  selections:
+    - anchor: {line: 0, character: 15}
+      active: {line: 0, character: 15}
+  marks: {}
+finalState:
+  documentContents: "lambda "
```

Comment:
I think the `lambda` should be included here

Reply:
Hey @pokey, good catch. What is happening here is that both `block - `.each { |x| ... ` and `do_block -`.each do |x| ...` are also defined as lambdas so there is some ambiguity here. I guess ideally, we'd pull the `->` and the `lambda` out in this case. It does seem a bit odd that the behavior is different between a `.each { |x|` where only the block is taken versus the lambda function where everything including the block is taken. I do think the behavior you're specifying though is what we want. 

`lambda` under `anonymousFunction: ["lambda", "block", "do_block"],` isn't doing anything right now, I guess I need to look at how to do a precedence match and if a block is preceded by a lambda, match both rather than just the block.

I'm using the [tree-sitter playground](https://tree-sitter.github.io/tree-sitter/playground) to generate these but here's what a lambda function looks like versus a regular block:

```
hi = -> { puts "Hi!" }
[program] [0, 0] - [1, 0]
  [assignment] [0, 0] - [0, 22]
    left: [identifier] [0, 0] - [0, 2]
    right: [lambda] [0, 5] - [0, 22]
      body: [block] [0, 8] - [0, 22]
        [call] [0, 10] - [0, 20]
          method: [identifier] [0, 10] - [0, 14]
          arguments: [argument_list] [0, 15] - [0, 20]
            [string] [0, 15] - [0, 20]
              [string_content] [0, 16] - [0, 19]
```

vs 

```
h.each do |x|
	puts x
end

[program] [0, 0] - [3, 0]
  [call] [0, 0] - [2, 3]
    receiver: [identifier] [0, 0] - [0, 1]
    method: [identifier] [0, 2] - [0, 6]
    block: [do_block] [0, 7] - [2, 3]
      parameters: [block_parameters] [0, 10] - [0, 13]
        [identifier] [0, 11] - [0, 12]
      [call] [1, 1] - [1, 7]
        method: [identifier] [1, 1] - [1, 5]
        arguments: [argument_list] [1, 6] - [1, 7]
          [identifier] [1, 6] - [1, 7]
```

Reply:
Actually got it working for one of them using `lambda?.do_block`

Reply:
Just pushed a fix.  Lmk if that makes sense https://github.com/cursorless-dev/cursorless-vscode/pull/578/commits/3e83c2d34e394373e18afb4c15e8b8304cfd85e2

Reply:
Yep, I think it makes sense. I need to dig into the why we don't need to specify `block` by itself in `cascadingMatcher` but I tested the cases and it looks like it is doing the right thing

Reply:
Block only ever appears as a child of a `lambda` or `block` node.  The former is handled by `"lambda"`, the latter by the new `blockMatcher`.  Make sense?

Reply:
Yep yep

Reply:
The reason I had to remove it is that otherwise it would have stopped the traversal code before it had a chance to see the parent `call` node